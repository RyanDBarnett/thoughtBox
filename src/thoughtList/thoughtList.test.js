import React from 'react';
import { shallow, mount } from 'enzyme';

import { ThoughtList } from './ThoughtList.js';

describe('ThoughtList', () => {

  it('should match the snapshot with no thoughts', () => {
    const wrapper = shallow(<ThoughtList thoughtList={[]} />);
    expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with thoughts', () => {
    const mockThoughts = [
      {
        title: 'Foo',
        body: 'Bar', 
        id: 0
      }, 
      {
        title: 'Baz',
        body: 'Boo', 
        id: 1
      }
    ];
    const wrapper = shallow(
      <ThoughtList thoughtList={mockThoughts} />
    );
    expect(wrapper).toMatchSnapshot()
  });

});
