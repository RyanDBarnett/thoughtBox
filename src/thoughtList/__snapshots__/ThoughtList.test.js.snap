// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThoughtList should match the snapshot with no thoughts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThoughtList
    thoughtList={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ThoughtList
        </h2>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ThoughtList",
        },
        "ref": null,
        "rendered": "ThoughtList",
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ThoughtList
          </h2>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ThoughtList",
          },
          "ref": null,
          "rendered": "ThoughtList",
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ThoughtList should match the snapshot with thoughts 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThoughtList
    thoughtList={
      Array [
        Object {
          "body": "Bar",
          "id": 0,
          "title": "Foo",
        },
        Object {
          "body": "Boo",
          "id": 1,
          "title": "Baz",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ThoughtList
        </h2>,
        Array [
          <ThoughtCard
            body="Bar"
            id={0}
            title="Foo"
          />,
          <ThoughtCard
            body="Boo"
            id={1}
            title="Baz"
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ThoughtList",
        },
        "ref": null,
        "rendered": "ThoughtList",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "body": "Bar",
          "deleteThought": undefined,
          "id": 0,
          "title": "Foo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "body": "Boo",
          "deleteThought": undefined,
          "id": 1,
          "title": "Baz",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ThoughtList
          </h2>,
          Array [
            <ThoughtCard
              body="Bar"
              id={0}
              title="Foo"
            />,
            <ThoughtCard
              body="Boo"
              id={1}
              title="Baz"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ThoughtList",
          },
          "ref": null,
          "rendered": "ThoughtList",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "body": "Bar",
            "deleteThought": undefined,
            "id": 0,
            "title": "Foo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "body": "Boo",
            "deleteThought": undefined,
            "id": 1,
            "title": "Baz",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
